test/distributed/fsdp/test_fsdp_dtensor_state_dict.py

```diff
@@ -227,6 +228,30 @@ def test_dtensor_sharded_model_load_state_dict(self, offload_to_cpu):
             self.assertEqual(v1, v2)
 
 
+class TestFSDPDeviceMeshInit(DTensorTestBase):
+    @with_comms
+    @skip_if_lt_x_gpu(2)
+    def test_fsdp_device_mesh_init_1d(self):
+        mesh_tensor = torch.arange(self.world_size)
+        device_mesh = DeviceMesh(self.device_type, mesh_tensor)
+        model = FSDP(TestDummyModel().cuda(), _device_mesh=device_mesh)
+        optim = torch.optim.Adam(model.parameters(), lr=0.1)
+        model(model.get_input()).sum().backward()
+
+    @with_comms
+    @skip_if_lt_x_gpu(2)
+    def test_fsdp_device_mesh_init_2d(self):
+        mesh_tensor = torch.arange(self.world_size).view(2, -1)
+        device_mesh = DeviceMesh(self.device_type, mesh_tensor)
+        model = FSDP(
+            TestDummyModel().cuda(),
+            _device_mesh=device_mesh,
+            sharding_strategy=ShardingStrategy.HYBRID_SHARD,
+        )
+        optim = torch.optim.Adam(model.parameters(), lr=0.1)
+        model(model.get_input()).sum().backward()
```

Comment:
nit (i know the PR is WIP) ideally the test should validate something like we're sharded / not sharded over the right dimensions?